viewing_party/party.py

```diff
@@ -1,23 +1,112 @@
-
+# WAVE 1
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if not title or not genre or not rating:
+        return None
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+def watch_movie(user_data, title):
+    for movies in user_data["watchlist"]:
+        if title == movies["title"]:
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
 
+# WAVE 2
+def get_watched_avg_rating(user_data):
+    ratings = 0.0
+    count_movies = 0
+    if not user_data["watched"]:
+        return ratings 
+    for movie_info in user_data["watched"]:
+        if isinstance(movie_info.get("rating"),float):
+            ratings += movie_info.get("rating")
+            count_movies += 1
+    return ratings/count_movies
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_most_watched_genre(user_data):
+    favorite_genre ={}
+    if not user_data["watched"]:
+        return None 
+    for movie_info in user_data["watched"]: 
+        if movie_info["genre"] not in favorite_genre:
+            favorite_genre[movie_info["genre"]] = 1
+        else:
+            favorite_genre[movie_info["genre"]] += 1
+    return max(favorite_genre, key=favorite_genre.get)        
+            
+# WAVE 3
+def get_unique_watched(user_data):
+    friends_watched_movie = set()
+    user_watched_movie = []
+    unique_movie_watched = []
+    # generate a set with movie titles from friend's watched lists
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
+            friends_watched_movie.add(movie["title"])
+   # generate a list of dictionaries with watched user movie info
+    for user_movies in user_data["watched"]:
+        user_watched_movie.append(user_movies)
+    # Check to see if the user movie title is within the whole set of friend's watched list
+    for movie_info in user_watched_movie:
+        if movie_info["title"] not in friends_watched_movie:
+            unique_movie_watched.append(movie_info)
+    return unique_movie_watched
 
+def get_friends_unique_watched(user_data): 
+    movie_friend_watched = []
+    only_friends_watched = []
+    for watched_list in user_data["friends"]:
+        for movie in watched_list["watched"]:
+            if movie not in movie_friend_watched:
+                movie_friend_watched.append(movie)
+    
+    for movie_info in movie_friend_watched:
+        if movie_info not in user_data["watched"]:

```

Comment:
What's the big-O time complexity of this loop? (Consider what `in` has to do in the `if` statement.)